### PR TITLE
[DS-3469] virus scan during submission attempts to read uploaded bitstream as anonymous user, which fails

### PR DESCRIPTION
**Pull request by mwoodiupui**
https://jira.duraspace.org/browse/DS-3469

This *untested* patch looks like it should supply the necessary identity to authorize the task (assuming that the current session *is* authorized).

### CODE REVIEW

**Comment by mwoodiupui** [comment]
Noted in passing:  WorkflowCuratorServiceImpl probably needs a similar treatment.

File: dspace-api/src/main/java/org/dspace/curate/Curator.java [hunk 1]
```diff
@@ -268,7 +268,22 @@ else if (type == Constants.SITE)
...
+     * (Community, Collection or Item).
+     *
+     * @param c session context in which curation takes place.
+     * @param dso the single object to be curated.
+     * @throws java.io.IOException passed through.
+     */
+    public void curate(Context c, DSpaceObject dso)
+            throws IOException
+    {
+        curationCtx.set(c);
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/curate/Curator.java]: Ugh, this has the side-effect of leaving the Curator set to the given Context.  Should we save the old reference and restore it before return?

Or should Curator have a setContext(Context) instead?

**Comment by hardyoyo** [comment]
testing this PR locally, it seems more stable. I'm assuming that the UI you're using is XMLUI, I'm using JSPUI and it doesn't give any feedback to what the error might be, just a little yellow exclamation point (see https://jira.duraspace.org/browse/DS-3375). However, uploads of non-virus files seem to work fine, even with the vscan curation task configured, which is good enough for me. I am not sure what Travis is complaining about, but, assuming that is a transient error, +1

**Review by hardyoyo — APPROVED** [review]
This all looks good, and passes my testing.

**Review by tdonohue — APPROVED** [review]
👍 Looks good and obvious (mostly minor code cleanup).

File: dspace-api/src/main/java/org/dspace/curate/Curator.java (hunk 1 above)
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/curate/Curator.java]: As discussed on IRC, I just added a note to the method doc comment.

**Pull request merged**